apps/users/views.py

```diff
@@ -1,109 +1,165 @@
+        else:
+            key_access_update_form = None
 
         if "save_profile" in request.POST:
-            if user_form.is_valid() and profile_form.is_valid():
-                user_form.save()
-                profile_form.save()
-                messages.success(request, "Your profile was successfully updated!")
+            if profile_update_form.is_valid():
+                profile_update_form.save()
+                messages.success(request, "Your profile has been updated!")
                 return redirect(reverse("cosmos_users:user_profile") + "#profile")
         elif "save_password" in request.POST:
-            if password_form.is_valid():
-                password_form.save()
-                messages.success(request, "Your password was succesfully updated!")
+            if password_change_form.is_valid():
+                password_change_form.save()
+                messages.success(request, "Your password has been updated!")
                 return redirect(reverse("cosmos_users:user_profile") + "#password")
         elif "save_preferences" in request.POST:
-            if profile_form.is_valid():
-                profile_form.save()
-                messages.success(request, "Your preferences were succesfully updated!")
+            if preferences_update_form.is_valid():
+                preferences_update_form.save()
+                messages.success(request, "Your preferences have been updated!")
                 return redirect(reverse("cosmos_users:user_profile") + "#preferences")
         elif "save_key_access" in request.POST:
-            if profile_form.is_valid():
-                profile_form.save()
-                messages.success(request, "Your key access settings were succesfully updated!")
+            if key_access_update_form is not None and key_access_update_form.is_valid():
+                key_access_update_form.save()
+                messages.success(request, "Your key access settings have been updated!")
                 return redirect(reverse("cosmos_users:user_profile") + "#key-access")
 
-        if user_form.is_valid() and profile_form.is_valid():
-            user_form.save()
-            profile_form.save()
-            messages.success(request, "Your profile was successfully updated!")
-            return redirect("/")
-        else:
-            messages.error(request, "Please correct the error below.")
     else:
-        user_form = MemberUpdateForm(instance=request.user)
-        profile_form = ProfileUpdateForm(instance=request.user.profile)
-        password_form = PasswordChangeForm(user=request.user)
+        profile_update_form = ProfileUpdateForm(instance=request.user)
+        password_change_form = PasswordUpdateForm(user=request.user)
+        preferences_update_form = PreferencesUpdateForm(instance=request.user.profile)
+
+        if request.user.profile.institution_name == "tue":
```

Comment:
we have way too many ways to derive which institution a profile is associated with. we need to standardize it.